File: app/assets/javascripts/delivery-method.js
$(function(){
  // 配送方法ボックスのオプションを作成
  function appendOption(method){
    var html = `<option value="${method.id}">${method.name}</option>`;
    return html;
  }
// 配送料の負担を選択で配送方法を表示させる
  $('#paying-side-selection').change(function () {
    var paySide_id = $('#paying-side-selection option:selected').val();
    if (paySide_id.length != 0){
      $('#delivery-method-box').removeClass('hide');
      $.ajax({
        url: 'get_delivery_method',
        type: 'Get',
        data: {paySide_id: paySide_id},
        dataType: 'json'
      }).done(function(delivery_methods){
        $('#delivery-method-selection option').remove();
        var insertHTML = `<option value="">---</option>`;
        delivery_methods.forEach(function(delivery_method){
          insertHTML += appendOption(delivery_method);
        })
        $('#delivery-method-selection').append(insertHTML);        
      }).fail(function(){
        alert('配送方法の取得に失敗しました');
      })
    } else {
      $('#delivery-method-box').addClass('hide');
    }
  });
});

File: app/models/delivery_method.rb
class DeliveryMethod < ActiveHash::Base
  # id=1~9は送料込み・出品者負担
  # id= 10~13は着払い・購入者負担
  self.data = [
    {id: 1, name: '未定', payside_id: 1}, {id: 2, name: 'らくらくメルカリ便', payside_id: 1}, {id: 3, name: 'ゆうメール', payside_id: 1}, {id: 4, name: 'レターパック', payside_id: 1}, {id: 5, name: '普通郵便(定形、定形外)', payside_id: 1}, {id: 6, name: 'クロネコヤマト', payside_id: 1},
    {id: 7, name: 'ゆうパック', payside_id:1}, {id: 8, name: 'クリックポスト', payside_id: 1}, {id: 9, name: 'ゆうパケット', payside_id: 1},
    {id: 10, name: '未定', payside_id: 2}, {id: 11, name: 'クロネコヤマト', payside_id: 2}, {id: 12, name: 'ゆうパック', payside_id: 2}, {id: 13, name: 'ゆうメール', payside_id: 2}
  ]
end